framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -875,75 +874,77 @@ public void typeProcess(TypeElement e, TreePath p) {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// Reporting type-checking errors; the main entry point is report()
+    /// Reporting type-checking errors; most clients use reportError() or reportWarning()
     ///
 
     /**
-     * Reports a result. By default, it prints it to the screen via the compiler's internal messager
-     * if the result is non-success; otherwise, the method returns with no side effects.
+     * Reports an error. By default, prints it to the screen via the compiler's internal messager.
      *
-     * @param r the result to report
-     * @param src the position object associated with the result; may be an Element, a Tree, or null
+     * @param source the source position information; may be an Element, a Tree, or null
+     * @param messageKey the message key
+     * @param args arguments for interpolation in the string corresponding to the given message key
      */
-    public void report(final Result r, final Object src) {
-        if (r.isSuccess()) {
-            return;
-        }
+    public void reportError(Object source, @CompilerMessageKey String messageKey, Object... args) {
```

Comment:
I'm using the Result class in #2883 to collect multiple errors and report them once:
https://github.com/typetools/checker-framework/pull/2883/files#diff-31033326203125d5a0af0323436d39f4R463

So, I would like to keep the class.  

I support changing the report methods so that you don't have to create a Result object only to pass it to this method.  I do think we should deprecate the previous methods though as this is a rather large breaking change.

Reply:
@smillst I deprecated the Result class instead of removing it.

Instead of `Result`, use `List<DiagMessage>`.  (Currently a `Result` is essentially just a `List<DiagMessage>` plus some baggage.)

The refactoring is easy.  It's two commands in Emacs; I will make a pull request for you.  I'm happy to do it for any other branches or forks where people are concerned about merge conflicts.